src/csim/stat_ops_expectation_value.cpp

```diff
@@ -260,15 +823,40 @@ double expectation_value_multi_qubit_Pauli_operator_partial_list_single_thread(
         Pauli_operator_type_list, target_qubit_index_count, &bit_flip_mask,
         &phase_flip_mask, &global_phase_90rot_count, &pivot_qubit_index);
     double result;
+
+#ifdef _USE_SVE
+    // # of complex128 numbers in an SVE register
+    ITYPE VL = svcntd() / 2;
+#endif
+
     if (bit_flip_mask == 0) {
-        result =
-            expectation_value_multi_qubit_Pauli_operator_Z_mask_single_thread(
-                phase_flip_mask, state, dim);
+#ifdef _USE_SVE
+        if (dim > VL) {
+            result =
+                expectation_value_multi_qubit_Pauli_operator_Z_mask_single_thread_sve(
+                    phase_flip_mask, state, dim);
+        } else
+#endif
+        {
+            result =
+                expectation_value_multi_qubit_Pauli_operator_Z_mask_single_thread(
+                    phase_flip_mask, state, dim);
+        }
     } else {
-        result =
-            expectation_value_multi_qubit_Pauli_operator_XZ_mask_single_thread(
-                bit_flip_mask, phase_flip_mask, global_phase_90rot_count,
-                pivot_qubit_index, state, dim);
+#ifdef _USE_SVE
+        if ((dim >> 1) > VL) {
```

Comment:
`expectation_value_multi_qubit_Pauli_operator_XZ_mask_single_thread_sve()` also processes VL<<1 elements in one step, right?